src/scrobbler/subsonicscrobbler.cpp

```diff
@@ -0,0 +1,111 @@
+/*
+ * Strawberry Music Player
+ * Copyright 2018, Jonas Kvinge <jonas@jkvinge.net>
+ * Copyright 2020, Pascal Below <spezifisch@below.fr>
+ *
+ * Strawberry is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Strawberry is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Strawberry.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "config.h"
+
+#include <QtGlobal>
+#include <QVariant>
+#include <QString>
+#include <QDateTime>
+#include <QTimer>
+#include <QtDebug>
+
+#include "core/application.h"
+#include "core/song.h"
+#include "core/timeconstants.h"
+#include "core/logging.h"
+#include "core/closure.h"
+#include "internet/internetservices.h"
+#include "subsonic/subsonicservice.h"
+
+#include "audioscrobbler.h"
+#include "scrobblerservice.h"
+#include "subsonicscrobbler.h"
+
+const char *SubsonicScrobbler::kName = "Subsonic";
+
+SubsonicScrobbler::SubsonicScrobbler(Application *app, QObject *parent) : ScrobblerService(kName, app, parent),
+  app_(app),
+  service_(app->internet_services()->Service<SubsonicService>()),
+  submitted_(false) {
+  ReloadSettings();
+}
+
+SubsonicScrobbler::~SubsonicScrobbler() {
+
+}
+
+void SubsonicScrobbler::ReloadSettings() {
+
```

Comment:
You can read serversidescrobbling from the subsonic settings here and set enabled_ based on that so that it can be used by AudioScrobbler.